src/spatialdata_plot/pl/utils.py

```diff
@@ -1031,3 +1051,186 @@ def _mpl_ax_contains_elements(ax: Axes) -> bool:
+        )
+
+    return image
+
+
+def _multiscale_to_spatial_image(
+    multiscale_image: MultiscaleSpatialImage,
+    element: str,
+    dpi: float,
+    width: float,
+    height: float,
+    scale: str | None = None,
+    is_label: bool = False,
+) -> SpatialImage:
+    """Extract the SpatialImage to be rendered from a multiscale image.
+
+    From the `MultiscaleSpatialImage`, the scale that fits the given image size and dpi most is selected
+    and returned. In case the lowest resolution is still too high, a rasterization step is added.
+
+    Parameters
+    ----------
+    multiscale_image
+        `MultiscaleSpatialImage` that should be rendered
+    element
+        name of the multiscale image
+    dpi
+        dpi of the target image
+    width
+        width of the target image in inches
+    height
+        height of the target image in inches
+    scale
+        specific scale that the user chose, if None the heuristic is used
+    is_label
+        When True, the multiscale image contains labels which don't contain the `c` dimension
+
+    Returns
+    -------
+    SpatialImage
+        To be rendered, extracted from the MultiscaleSpatialImage respecting the dpi and size of the target image.
+    """
+    scales = [leaf.name for leaf in multiscale_image.leaves]
+    x_dims = [multiscale_image[scale].dims["x"] for scale in scales]
+    y_dims = [multiscale_image[scale].dims["y"] for scale in scales]
+
+    if isinstance(scale, str):
+        if scale not in scales and scale != "full":
+            raise ValueError(f'Scale {scale} does not exist. Please select one of {scales} or set scale = "full"!')
+        optimal_scale = scale
+        if scale == "full":
+            # use scale with highest resolution
+            optimal_scale = scales[np.argmax(x_dims)]
+    else:
+        # ensure that lists are sorted
+        order = np.argsort(x_dims)
+        scales = [scales[i] for i in order]
+        x_dims = [x_dims[i] for i in order]
+        y_dims = [y_dims[i] for i in order]
+
+        optimal_x = width * dpi
+        optimal_y = height * dpi
+
+        # get scale where the dimensions are close to the optimal values
+        # when possible, pick higher resolution (worst case: downscaled afterwards)
+        optimal_index_y = np.searchsorted(y_dims, optimal_y)
+        if optimal_index_y == len(y_dims):
+            optimal_index_y -= 1
+        optimal_index_x = np.searchsorted(x_dims, optimal_x)
+        if optimal_index_x == len(x_dims):
+            optimal_index_x -= 1
+
+        # pick the scale with higher resolution (worst case: downscaled afterwards)
+        optimal_scale = scales[min(optimal_index_x, optimal_index_y)]
+
+    # NOTE: problematic if there are cases with > 1 data variable
+    data_var_keys = list(multiscale_image[optimal_scale].data_vars)
+    image = multiscale_image[optimal_scale][data_var_keys[0]]
+
+    logging.info(f"Using scale '{optimal_scale}' from multi-scale image '{element}'.")
```

Comment:
Same as above, I would remove this. The user will get the plot fast, this message can't be used by the user in a actionable way. I would still highlight this in the documentation notebook.

Reply:
Yeah, I get your point. Was probably most useful for me to see what was going on during coding and to demonstrate it in the documentation nb. It might help the user to see which scale is selected per default so they could better judge which scales might be too large. But would probably be rarely used